README.md

```diff
@@ -2,31 +2,322 @@
+- Use the debugger to demonstrate polymorphism and dynamic binding
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+An **Interface** in Java is defined as an abstract type
+used to specify the behavior of a class.
+Interfaces are used to achieve abstraction, specifying
+what a class must do and not how. 
 
-## Topic 1
+## Code-Along
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Fork and clone this lesson.  The code includes classes `Main`,
+`Rectangle` and `Circle`.  You will create an interface named `Shape`
+and adapt the classes to implement the interface.
 
-## Topic 2
+## Motivation
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Let's look at an example application that demonstrates why interfaces are useful.
+Assume we own a restaurant with several rectangular tables, and we
+need to repaint the table tops.  We need to figure out the total
+area and circumference to determine how much paint to buy.
+
+The `Rectangle` class has methods to calculate area and circumference:
+
+```java
+public class Rectangle {
+    private double height, width;
+
+    public Rectangle(double height, double width) {
+        this.height = height;
+        this.width = width;
+    }
+
+    public double getArea() { return height * width; }
+    public double getCircumference () { return 2 * (height + width); }
+    
+}
+```
+
+The driver class `Main` has a  `main` method to
+store an array of `Rectangle` objects representing the table tops.
+It is easy to loop through the array and accumulate the total area
+and circumference of the rectangles.
+
+```java
+public class Main {
+    public static void main(String[] args) {
+        Rectangle[] tables = new Rectangle[] {
+                new Rectangle(7, 5),
+                new Rectangle(2, 4),
+                new Rectangle(3, 5)
+        };
+
+        double totalArea = 0;
+        double totalCircumference = 0;
+        for (Rectangle table: tables) {
+            totalArea += table.getArea();
+            totalCircumference += table.getCircumference();
+        }
+
+        System.out.println("Total area = " + totalArea);
+        System.out.println("Total circumference = " + totalCircumference);
+    }
+}
+
+```
+
+The program prints the totals:
+
+```text
+Total area = 58.0
+Total circumference = 52.0
+```
+
+But what if we decide to purchase two additional tables that are round rather than
+rectangular?    The `Circle` class has methods to calculate
+the area and circumference:
+
+```java
+public class Circle {
+    private double radius;
+
+    public Circle(double radius) {this.radius = radius;}
+    public double getArea() { return Math.PI * radius * radius; }
+    public double getCircumference () { return 2 * Math.PI * radius; }
```

Comment:
There looks to be an extra space between `getCircumference` and `()`

Reply:
fixed